ipld/encoding/src/cbor.rs

```diff
@@ -9,26 +9,10 @@ use std::rc::Rc;
 use serde::{Deserialize, Serialize};
 
 use super::errors::Error;
-use crate::{de, from_slice, ser, strict_bytes, to_vec};
+use crate::{de, from_slice, strict_bytes, to_vec};
 
 pub const DAG_CBOR: u64 = 0x71;
 
-/// Cbor utility functions for serializable objects
-pub trait Cbor: ser::Serialize + de::DeserializeOwned {
```

Comment:
Let's deprecate this in the encoding crate, but otherwise leave it (and all the impls in the encoding crate) alone. Otherwise, we need to release:

1. A minor (breaking) version of the endcoding crate.
2. A minor (breaking) version of all the crates that depend on it.

Basically, it makes a bunch of interfaces incompatible.

Alternatively, if you're willing to cut the releases and update everything, go for it. But it's not going to be a quick fix.

Reply:
Note: definitely remove the implementations from _other_ crates, I'd just leave the encoding crate alone for now (except deprecating it).